src/kem/kyber768.rs

```diff
@@ -52,11 +52,16 @@ pub fn generate_keypair(
     let ind_cpa_keypair_randomness = &randomness[0..parameters::CPA_PKE_KEY_GENERATION_SEED_SIZE];
     let implicit_rejection_value = &randomness[parameters::CPA_PKE_KEY_GENERATION_SEED_SIZE..];
 
-    let ind_cpa_key_pair = ind_cpa::generate_keypair(ind_cpa_keypair_randomness)?;
+    let (ind_cpa_key_pair, sampling_a_error) =
+        ind_cpa::generate_keypair(ind_cpa_keypair_randomness);
 
     let secret_key_serialized = ind_cpa_key_pair.serialize_secret_key(implicit_rejection_value);
 
-    Ok((ind_cpa_key_pair.pk(), secret_key_serialized))
+    if sampling_a_error.is_some() {
```

Comment:
Is this not going through hax?
Would allow you to avoid the `unwrap` and much nicer.

```suggestion
    `if let Some(error) = sampling_a_error {
```